index.js

```diff
@@ -70,6 +74,19 @@ module.exports = class Hyperswarm extends EventEmitter {
     this.dht.on('network-change', this._handleNetworkChange.bind(this))
   }
 
+  _shouldUseRelay () {
+    return this.relayThrough && this.dht.host && (this.dht.port === 0)
```

Comment:
`return this.relayThrough && this.dht.host?.port === 0` ?

Reply:
Doesn't actually need to be a function, can just be a const in _relayOptions, that might be clearer:

```
  _relayOptions () {
    const shouldUseRelay = this.relayThrough && this.dht.host?.port === 0 ?

    // Only provide relay options if we're not randomized and a relay address is available
    // TODO: Always holepunch once connection switchover is implemented
    return {
      relayThrough: () => shouldUseRelay ? this.relayThrough : null,
      holepunch: () => !shouldUseRelay
    }
  }     
```

Could also easily make _relayOptions a getter instead of a function

Reply:
Ya that'd also be fine

Reply:
Think @LuKks extended this branch, so might not be relevant anymore

Reply:
Yeah, these comments are a bit outdated, it's good as-is as far as I can see. IIRC Lukas said that it has to be a function because it either overrides a function or is overriden somewhere else